test/test-viz.cpp

```diff
@@ -0,0 +1,311 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <tuple>
+#include <variant>
+
+#include <lorina/bench.hpp>
+
+#define SDL_MAIN_HANDLED
+
+#include <SDL.h>
+
+enum SPACING {
+    HSPACING = 2,
+    VSPACING = 2
+};
+
+enum ERROR_CODES {
+    EMPTY_FILENAME = -1,
+    PARSER_FAILURE = -2,
+    SDL_INIT_FAILURE = -3
+};
+
+struct Vertex {
+    std::string type;
+    std::string name;
+    int layer;
+    std::vector<int> ins;
+    SDL_FRect position;
+};
+
+class Line {
+public:
+    explicit Line(const Vertex& first, const Vertex& second) {
+        x1 = first.position.x + first.position.w;
+        y1 = first.position.y + first.position.h/2;
+        x2 = second.position.x;
+        y2 = second.position.y + second.position.h/2;
+    }
```

Comment:
`Vertex` is not actually required, the calculation is performed on rectangles.

Reply:
Unfortunate naming: line is probably... well, a line, but this is calculated based on a rectangle bounds. Assuming there are 9 predefined rectangle coordinates (top, bottom, left, right, center, top-left, top-right, bottom-left, bottom-right), this is a right-to-left connection and could look like this:
```suggestion
    static Line rightToLeftLine(const SDL_FRect &src, const SDL_FRect &dst) {
        double x1, y1, x2, y2;
        getRight(src, x1, y1);
        getLeft(dst, x2, y2);
        return { x1, y1, x2, y2 };
    }
    
    static void getRight(const SDL_FRect &rect, double &x, double &y) {
        x = rect.x + rect.w;
        y = rect.y + rect.h / 2;
    }
    
    static void getLeft(const SDL_FRect &rect, double &x, double &y) {
        x = rect.x;
        y = rect.y + rect.h / 2;
    }

```